core/src/main/java/org/kohsuke/stapler/RequestImpl.java

```diff
@@ -1158,6 +1166,7 @@ private boolean isMultipart() {
     }
 
     @Override
+    @WithBridgeMethods(value = org.apache.commons.fileupload.FileItem.class, adapterMethod = "fromFileUpload2FileItem")
     public FileItem getFileItem(String name) throws ServletException, IOException {
```

Comment:
Maybe need deprecated this method and add a new method `getFileItem2` to return a `org.apache.commons.fileupload2.core.FileItem` Object.

Like we do for `FilePath.copyFrom`
https://github.com/jenkinsci/jenkins/pull/9259/files#diff-c1b889a6e9863d548250382cfc6e459ff2ace03c02640ec41c337026384045d8R1187

Otherwise, how can call this method when the plugin has been upgraded to use the fileupload2 api?

If I understand something wrong here, please tell me.

Reply:
> If I understand something wrong here, please tell me.

You are understanding something wrong here.